src/Elements/Comment.react.js

```diff
@@ -38,16 +57,111 @@ class Comment extends Component {
+                                    <a
+                                        className="cancel-link"
+                                        onClick={this._onCancelEditClick}
+                                        style={{ marginRight: '11px' }}
+                                    >
+                                        {'Cancel'}
+                                    </a>
+                                    <a
+                                        className="save-link"
+                                        disabled={!canEdit}
+                                        onClick={this._onSaveClick}
+                                        style={{ marginRight: '11px' }}
+                                    >
+                                        {'Save'}
+                                    </a>
+                                    <a
+                                        className="remove-link"
+                                        disabled={!canRemove}
+                                        onClick={this._onRemoveClick}
+                                    >
+                                        {'Remove'}
+                                    </a>
+                                </div>
+                            ) : (
+                                <div
+                                    style={{
+                                        flex: '1 0 auto',
+                                        margin: detailsPosition === 'right' ? '0 0 0 11px' : '0 11px 0 0',
+                                        textAlign: detailsPosition === 'right' ? 'left' : 'right'
+                                    }}
+                                >
+                                    <a
+                                        className="edit-link"
+                                        onClick={this._onEditClick}
+                                    >
+                                        {'Edit'}
+                                    </a>
+                                </div>
+                                ) :
+                            null
+                        }
                     </div>
                 ) : null}
 
                 <div className="comment-bubble">
-                    {children}
+                    {isEditable && canEdit && isEditMode ? (
+                        <TextArea
+                            fluid
+                            onChange={this._onCommentChange}
+                            onKeyDown={this._onEditKeyDown}
+                            ref={ref => this.editableCommentTextArea = ref}
+                            value={updatedCommentText}
+                        />
+                    ): children}
                 </div>
             </div>
         );
     }
 
+    _onCancelEditClick() {
+        this.setState({ isEditMode: false });
+    }
+
+    _onCommentChange(value) {
+        this.setState({ updatedCommentText: value });
+    }
```

Comment:
@morethanfire ... I wasn't able to get the Text Area to work here as an "uncontrolled" component.  I needed to initialize the Text Area's value and then do the standard `onChange` deal.  Note that I'm holding the "dirty" unsaved but updated comment text in state in here.  Maybe you and I can play with it Monday and make it better.